swift/ql/src/queries/Security/CWE-020/MissingRegexAnchor.qhelp

```diff
@@ -0,0 +1,74 @@
+<!DOCTYPE qhelp PUBLIC
+"-//Semmle//qhelp//EN"
+"qhelp.dtd">
+<qhelp>
+
+	<overview>
+		<p>
+
+			Sanitizing untrusted input with regular expressions is a
+			common technique.  However, it is error-prone to match untrusted input
+			against regular expressions without anchors such as <code>^</code> or
+			<code>$</code>.  Malicious input can bypass such security checks by
+			embedding one of the allowed patterns in an unexpected location.
```

Comment:
```suggestion
			Sanitizing untrusted input with regular expressions is a
			common technique, but malicious actors may be able to embed one of the 
			allowed patterns in an unexpected location. To prevent against this,
			you should use anchors in your regular expressions,
			such as <code>^</code> or <code>$</code>.
```